sklearn/tests/test_pipeline.py

```diff
@@ -330,6 +332,19 @@ def test_pipeline_methods_pca_svm():
     pipe.score(X, y)
 
 
+@pytest.mark.filterwarnings('ignore: default contamination parameter')  # 0.22
+def test_pipeline_score_samples_pca_lof():
+    iris = load_iris()
+    X = iris.data
+    # Test with PCA + OneClassSVM
+    clf = LocalOutlierFactor(novelty=True)
+    pca = PCA(svd_solver='full', n_components='mle', whiten=True)
+    pipe = Pipeline([('pca', pca), ('lof', clf)])
+    pipe.fit(X)
+    assert_allclose(pipe.score_samples(X[:2]),
```

Comment:
I find this assert a bit arbitrary.

Instead, I would test on the full X and check the shape and not the number inside.
If we want to check the values, we should probably make the manual transform by hand and check that we get the same number

```python
lof.fit(pca.fit_transform(X))
assert_allclose(pipe.score_samples(X), lof.score_samples(pca.transform(X))
```

Reply:
Which solution do you prefer ? Checking only the shapes, or checking the values with manual transform ?

Reply:
Sorry I did not edit my message. We can do both :)

Reply:
Thanks. I have updated the test.